code/modules/mob/living/carbon/xenomorph/abilities.dm

```diff
@@ -1195,15 +1195,38 @@
 	if(SSpoints.xeno_points_by_hive[X.hivenumber] < psych_cost)
 		to_chat(owner, "<span class='xenowarning'>Someone used all the psych points while we were building!</span>")
 		return fail_activate()
+	building_turret(A, X)
+	SSpoints.xeno_points_by_hive[X.hivenumber] -= psych_cost
+	succeed_activate()
 
+/datum/action/xeno_action/activable/build_turret/proc/building_turret(atom/A, mob/living/carbon/xenomorph/X)
 	to_chat(owner, "<span class='xenowarning'>We build a new acid turret, spending 100 psychic points in the process</span>")
 	new /obj/structure/xeno/resin/xeno_turret(get_turf(A), X.hivenumber)
-
-	SSpoints.xeno_points_by_hive[X.hivenumber] -= psych_cost
 	log_game("[owner] built a turret in [AREACOORD(A)], spending [psych_cost] psy points in the process")
 	xeno_message("[X.name] has built a new turret at [get_area(A)]!", "xenoannounce", 5, X.hivenumber)
 
-	succeed_activate()
+/////////////////////////////////
+/// Build xeno larva jelly turret
+/////////////////////////////////
+
+/datum/action/xeno_action/activable/build_turret/jelly
+	name = "Secrete larva jelly turret"
+	action_icon_state = "xeno_turret"
+	mechanics_text = "Creates a new xeno larva jelly turret for 150 points"
+	ability_name = "secrete larva jelly turret"
+
+/datum/action/xeno_action/activable/build_turret/jelly/can_use_ability(atom/A, silent, override_flags, mob/living/carbon/xenomorph/X, turf/T)
+	. = ..()
+	if(!.)
+		return FALSE
+	if(!T.check_alien_construction(X, planned_building = /obj/structure/xeno/resin/xeno_turret/jelly) || !T.check_disallow_alien_fortification(X))
+		return FALSE
```

Comment:
The `check_disallow_alien_fortification()` can be removed here, since it's already checked just above on the parent:
https://github.com/ProbablySpaghetti/XRF-13/blob/8d2f5f1106b0986f83e628ec3940cf91914bc4ba/code/modules/mob/living/carbon/xenomorph/abilities.dm#L1179-L1180
```suggestion
	if(!T.check_alien_construction(X, planned_building = /obj/structure/xeno/resin/xeno_turret/jelly))
		return FALSE
```